extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/ArcConstraintValidatorFactoryImpl.java

```diff
@@ -0,0 +1,40 @@
+package io.quarkus.hibernate.validator.runtime;
+
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorFactory;
+
+import org.hibernate.validator.internal.util.privilegedactions.NewInstance;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+
+public class ArcConstraintValidatorFactoryImpl implements ConstraintValidatorFactory {
+
+    @Override
+    public final <T extends ConstraintValidator<?, ?>> T getInstance(Class<T> key) {
+        InstanceHandle<T> instanceHandle = Arc.container().instance(key);
+        if (instanceHandle.isAvailable()) {
+            return instanceHandle.get();
+        } else {
+            return run(NewInstance.action(key, "ConstraintValidator"));
+        }
+    }
+
+    @Override
+    public void releaseInstance(ConstraintValidator<?, ?> instance) {
+        // we let ArC release the bean instances on shutdown as they were instantiated by ArC.
```

Comment:
This is OK for scoped validators, but if a validator is of `@Dependent` scope then ARC will produce a new validator instance for every call of `ArcConstraintValidatorFactoryImpl#getInstance` leading to a potential memory leak. Please consider this solution

```
    private final Map<Object, InstanceHandle<?>> destroyableValidators = new ConcurrentHashMap<>();
    
    @Override
    public <T extends ConstraintValidator<?, ?>> T getInstance(Class<T> key) {
        InstanceHandle<T> handle = Arc.container().instance(key);
        if (handle.isAvailable()) {
            T instance = handle.get();
            if (handle.getBean().getScope().equals(Dependent.class)) {
                destroyableValidators.put(instance, handle);
            }
            return instance;
        }
        return run( NewInstance.action( key, "ConstraintValidator" ) );
    }

    @Override
    public void releaseInstance(ConstraintValidator<?, ?> instance) {
        ofNullable(destroyableValidators.remove(instance)).ifPresent(InstanceHandle::destroy);
    }
    
    private <T> T run(PrivilegedAction<T> action) {
        return System.getSecurityManager() != null ? AccessController.doPrivileged( action ) : action.run();
    }
```

Reply:
I must admit that I only considered application scoped and singleton beans.

I added a commit along the lines of what you suggested.